love/src/components/Facility/FacilityMap.jsx

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react';
+import { defaultNumberFormatter } from 'Utils';
+import { M1M3ActuatorPositions } from 'Config';
+import styles from './FacilityMap.module.css';
+import Badge from '../GeneralPurpose/Badge/Badge';
+import Map from './Map/Map.jsx';
+import Device from './Map/Device.jsx';
+import EyeIcon from '../icons/EyeIcon/EyeIcon';
+import SimpleArrowIcon from '../icons/SimpleArrowIcon/SimpleArrowIcon';
+
+export default class FacilityMap extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      showHVAC: true,
+      showPower: true,
+      showMenu: true,
+      arrowDirection: 'right',
+      alarms: [],
+    };
+  }
+
+  componentDidMount = () => {
+    this.props.subscribeToStreams();
+  };
+
+  componentWillUnmount() {
+    this.props.unsubscribeToStreams();
+  }
+
+  getEyeIcon = (active) => {
+    return <EyeIcon active={!active} />;
+  };
+
+  hideHVAC = () => {
+    this.setState((prevState) => ({
+      showHVAC: !prevState.showHVAC,
+    }));
+  };
+
+  toggleLeftMenu = () => {
+    this.setState((prevState) => ({
+      showMenu: !prevState.showMenu,
+    }));
+    this.state.arrowDirection === 'right'
+      ? (this.state.arrowDirection = 'left')
+      : (this.state.arrowDirection = 'right');
+  };
+
+  hidePower = () => {
+    /*
+    this.setState(prevState => ({
+      showPower: !prevState.showPower
+    }));*/
+  };
+
+  render() {
+    const showHVAC = this.state.showHVAC;
+    const showPower = this.state.showPower;
```

Comment:
I suggest to use _object destructuring_, i.e. `const { showHVAC, showPower } = this.state;`